lib/puppet/util/run_mode.rb

```diff
@@ -55,11 +55,19 @@ def which_dir( system, user )
 
     class UnixRunMode < RunMode
       def conf_dir
-        which_dir("/etc/puppet", "~/.puppet")
+        which_dir("/etc/puppetlabs/agent", "~/.puppet")
       end
 
       def var_dir
-        which_dir("/var/lib/puppet", "~/.puppet/var")
+        which_dir("/opt/puppetlabs/agent/cache", "~/.puppet/var")
+      end
+
+      def run_dir
+        which_dir("/var/run/puppetlabs", "~/.puppet/var/run")
+      end
+
+      def log_dir
+        which_dir("/var/log/puppetlabs/agent", "~/.puppet/var/log/")
```

Comment:
These changes mean that I now have to specify --logdir and --rundir when starting the master because they no longer properly land under --vardir since this UnixRunMode subclass overrides the parent default value of $vardir/{log,run}.

Is this behavior desirable?

I see this topic has come up a bit already on this PR...